doc/introduction/interfaces/jax.rst

```diff
@@ -0,0 +1,137 @@
+.. _jax_interf:
+
+JAX interface
+=================
+
+In order to use PennyLane in combination with JAX, we have to generate JAX-compatible
+quantum nodes. A basic ``QNode`` can be translated into a quantum node that interfaces with JAX by using the ``interface='jax'`` flag in the QNode decorator.
+
+.. note::
+    
+    Currently, only the ``default.qubit.jax`` device supports the JAX interface.
+
+
+.. note::
+
+    To use the JAX interface in PennyLane, you must first
+    install ``jax`` and ``jaxlib``. You can then import PennyLane and JAX as follows:
+
+    .. code::
+
+        import pennylane as qml
+        import jax
+        import jax.numpy as jnp
+
+
+Construction via the decorator
+------------------------------
+
+The :ref:`QNode decorator <intro_vcirc_decorator>` is the recommended way for creating
+a JAX-capable QNode in PennyLane. Simply specify the ``interface='jax'`` keyword argument:
+
+.. code-block:: python
+    
+    dev = qml.device('default.qubit.jax', wires=2)
+
+    @qml.qnode(dev, interface='jax')
+    def circuit1(phi, theta):
+        qml.RX(phi[0], wires=0)
+        qml.RY(phi[1], wires=1)
+        qml.CNOT(wires=[0, 1])
+        qml.PhaseShift(theta, wires=0)
+        return qml.expval(qml.PauliZ(0)), qml.expval(qml.Hadamard(1))
+
+The QNode ``circuit1()`` is now a JAX-capable QNode, accepting ``jax.DeviceArray`` objects
+as input, and returning ``jax.DeviceArray`` objects. It can now be used like any other JAX function:
+
+>>> phi = jnp.array([0.5, 0.1])
+>>> theta = jnp.array(0.2)
+>>> circuit1(phi, theta)
+DeviceArray([0.8776, 0.6880], dtype=float64)
+
+Quantum gradients using JAX
+---------------------------
+
+Since a JAX-interfacing QNode acts like any other JAX interfacing python function,
+the standard method used to calculate gradients with JAX can be used.
+
+For example:
+
+.. code-block:: python
+
+    dev = qml.device('default.qubit.jax', wires=2)
+
+    @qml.qnode(dev, interface='jax')
+    def circuit3(phi, theta):
+        qml.RX(phi[0], wires=0)
+        qml.RY(phi[1], wires=1)
+        qml.CNOT(wires=[0, 1])
+        qml.PhaseShift(theta, wires=0)
+        return qml.expval(qml.PauliZ(0))
+
+    phi = jnp.array([0.5, 0.1])
+    theta = jnp.array(0.2)
+    grads = jax.grad(circuit3, argnums=(0, 1))
+    phi_grad, theta_grad = grads(phi, theta)
+
+This has output:
+
+>>> phi_grad
+DeviceArray([-0.47942555,  0.        ], dtype=float32)
+>>> theta_grad
+DeviceArray(-3.4332792e-10, dtype=float32)
+
+
+.. _jax_optimize:
+
+Using jax.jit on QNodes
+-----------------------
+
+To fully utilize the power and speed of JAX, you'll need to just-in-time compile
+your functions. If you're only taking expectation values, you can simply add the

```

Comment:
What about: probabilities, states, variances? Maybe it is useful to state the precise rule here, for example "if your circuit results are samples" or something...

Reply:
Yep, I was actually wondering that (e.g., returning expvals/probs with a finite number of shots)

Reply:
So the rule is that it's only used when this method is invoked.

https://github.com/PennyLaneAI/pennylane/blob/b16cb4382b230afb574beae6556302a4e081ae3b/pennylane/devices/default_qubit_jax.py#L219